listFiles mediator triggering errors when the directory is empty on the FTP

Hi Team,

When using the listFiles mediator to list the files in a FTP location in a sequence. & when the folder is empty it is spamming the following error

-06-13 14:05:24,376] ERROR {ListFiles} - {api:hello} Error while performing file:listFiles for folder ftp://cj:***@ftp.test.com:21/test_charith org.apache.commons.vfs2.FileSystemException: Could not find files in "ftp://cj:***@ftp.test.com/test_charith".
	at org.apache.commons.vfs2.provider.AbstractFileObject.findFiles(AbstractFileObject.java:957)
	at org.apache.commons.vfs2.provider.AbstractFileObject.listFiles(AbstractFileObject.java:1571)
	at org.apache.commons.vfs2.provider.AbstractFileObject.findFiles(AbstractFileObject.java:932)
	at org.wso2.carbon.connector.operations.ListFiles.getFilesAndFolders(ListFiles.java:285)
	at org.wso2.carbon.connector.operations.ListFiles.listFilesInHierarchicalFormat(ListFiles.java:218)
	at org.wso2.carbon.connector.operations.ListFiles.listFilesInFolder(ListFiles.java:188)
	at org.wso2.carbon.connector.operations.ListFiles.connect(ListFiles.java:124)
	at org.wso2.carbon.connector.core.AbstractConnector.mediate(AbstractConnector.java:32)
	at org.apache.synapse.mediators.ext.ClassMediator.updateInstancePropertiesAndMediate(ClassMediator.java:178)
	at org.apache.synapse.mediators.ext.ClassMediator.mediate(ClassMediator.java:97)
	at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:126)
	at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:74)
	at org.apache.synapse.mediators.template.TemplateMediator.mediate(TemplateMediator.java:147)
	at org.apache.synapse.mediators.template.InvokeMediator.mediate(InvokeMediator.java:180)
	at org.apache.synapse.mediators.template.InvokeMediator.mediate(InvokeMediator.java:96)
	at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:126)
	at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:74)
	at org.apache.synapse.mediators.base.SequenceMediator.mediate(SequenceMediator.java:158)
	at org.apache.synapse.mediators.base.SequenceMediator.mediate(SequenceMediator.java:214)
	at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:126)
	at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:74)
	at org.apache.synapse.mediators.base.SequenceMediator.mediate(SequenceMediator.java:158)
	at org.apache.synapse.api.Resource.process(Resource.java:342)
	at org.apache.synapse.api.API.process(API.java:477)
	at org.apache.synapse.api.AbstractApiHandler.apiProcess(AbstractApiHandler.java:94)
	at org.apache.synapse.api.AbstractApiHandler.dispatchToAPI(AbstractApiHandler.java:72)
	at org.apache.synapse.api.rest.RestRequestHandler.dispatchToAPI(RestRequestHandler.java:90)
	at org.apache.synapse.api.rest.RestRequestHandler.process(RestRequestHandler.java:76)
	at org.apache.synapse.rest.RESTRequestHandler.process(RESTRequestHandler.java:54)
	at org.apache.synapse.core.axis2.Axis2SynapseEnvironment.injectMessage(Axis2SynapseEnvironment.java:350)
	at org.apache.synapse.core.axis2.SynapseMessageReceiver.receive(SynapseMessageReceiver.java:101)
	at org.apache.axis2.engine.AxisEngine.receive(AxisEngine.java:180)
	at org.apache.synapse.transport.passthru.ServerWorker.processNonEntityEnclosingRESTHandler(ServerWorker.java:399)
	at org.apache.synapse.transport.passthru.ServerWorker.run(ServerWorker.java:213)
	at org.apache.axis2.transport.base.threads.NativeWorkerPool$1.run(NativeWorkerPool.java:172)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.apache.commons.vfs2.FileNotFolderException: Could not list the contents of "ftp://cj:***@test.com/test_charith" because it is not a folder.
	at org.apache.commons.vfs2.provider.AbstractFileObject.getChildren(AbstractFileObject.java:1047)
	at org.apache.commons.vfs2.provider.ftp.FtpFileObject.getChildren(FtpFileObject.java:357)
	at org.apache.commons.vfs2.provider.AbstractFileObject.traverse(AbstractFileObject.java:132)
	at org.apache.commons.vfs2.provider.AbstractFileObject.findFiles(AbstractFileObject.java:954)
	... 37 more


To reproduce the behaviour, simply use a listFiles operation against an empty folder in a FTP location.

Thanks,
Charith

--- a/commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/ftp/FtpFileObject.java
+++ b/commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/ftp/FtpFileObject.java
@@ -369,7 +369,7 @@ public class FtpFileObject extends AbstractFileObject<FtpFileSystem> {
         doGetChildren();
 
         // VFS-210
-        if (children.isEmpty()) {
+        if (children == null) {
             return null;
         } else if (children.isEmpty()) {
             return new String[0];

